test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1049,3 +1069,30 @@
                         (utils/get-extension-value utils/subject-alt-name-oid)
                         (:dns-name))]
       (is (= #{"puppet" "master"} (set alt-names))))))
+
+(deftest file-permissions
+  (let [tmpdir (System/getProperty "java.io.tmpdir")]
+    (testing "A newly created file contains the properly set permissions"
+      (doseq [u all-perms
+              g no-write-perms
```

Comment:
AHh yes, I probably should have documented this. On OS X, it won't actually allow me to set write perms for group or other.